Sources/ScipioKit/ProjectGenerator/ProjectGenerator.swift

```diff
@@ -0,0 +1,505 @@
+                return "\(packageName) is invalid package"
+            case .unsupportedTarget(let targetName, let kind):
+                return "\(targetName) is \(kind) but \(kind) is not supported yet"
+            case .localizationNotSupported(let targetName):
+                return "\(targetName) has localized resources but localized resources are not supported yet"
+            case .unknownError:
+                return "Unknown errors are occurred"
+            }
+        }
+    }
+
+    @discardableResult
+    private func addObject<T: PBXObject>(_ object: T, context: String? = nil) -> T {
+        pbxProj.add(object: object)
+        object.context = context
+        return object
+    }
+
+    private var sourceRoot: AbsolutePath? {
+        return package.graph.rootPackages.first?.path
+    }
+
+    func generate() throws {
+        let projectPath = package.projectPath
+        let parentDirectoryPath = package.projectPath.deletingLastPathComponent()
+
+        preparePBXProj()
+
+        guard let sourceRootDir = package.graph.rootPackages.first?.path else {
+            throw Error.invalidPackage(packageName: package.name)
+        }
+        pbxProj.rootObject?.projectDirPath = URL(fileURLWithPath: sourceRootDir.pathString, relativeTo: parentDirectoryPath).path
+
+        try generateTargets()
+
+        let projectFile = XcodeProj(workspace: .init(),
+                                    pbxproj: pbxProj)
+        try projectFile.write(pathString: projectPath.path, override: true)
+    }
+
+    private func generateTargets() throws {
+        // First, generate PBXTargets of all libraries
+        let targetsToGenerate = package.graph.reachableTargets
+            .filter { $0.type == .library }
+            .sorted { $0.name < $1.name }
+
+        let xcodeTargets: [ResolvedTarget: PBXNativeTarget] = try targetsToGenerate.reduce(into: [:]) { targets, target in
+            let xcodeTarget = addObject(
+                try makeTarget(for: target)
+            )
+            targets[target] = xcodeTarget
+        }
+        xcodeTargets.values.forEach { self.pbxProj.rootObject?.targets.append($0) }
+
+        // Generate ModuleMaps for Clang targets
+        let moduleMaps: [ResolvedTarget: AbsolutePath] = try xcodeTargets.reduce(into: [:]) { (collection, tuple) in
+            let (target, xcodeTarget) = tuple
+            if let clangTarget = target.underlyingTarget as? ClangTarget {
+                if let moduleMapPath = try applyClangTargetSpecificSettings(for: clangTarget, xcodeTarget: xcodeTarget).moduleMapPath {
+                    collection[target] = moduleMapPath
+                }
+            }
+        }
+
+        // Make LinkPhase for each Xcode targets
+        for (target, xcodeTarget) in xcodeTargets {
+            let dependsTargets = try target.recursiveDependencies().compactMap { value in
+                if case .target(let dependency, _) = value {
+                    return dependency
```

Comment:
`value` is ambiguous.
How about this change?
```suggestion
            let dependsTargets = try target.recursiveDependencies().compactMap { dependency in
                if case .target(let resolvedTarget, _) = dependency {
                    return resolvedTarget
```